pennylane/labs/resource_estimation/resource_container.py

```diff
@@ -238,6 +242,73 @@ def mul_in_parallel(first: Resources, scalar: int, in_place=False) -> Resources:
     return Resources(new_wires, new_gates, new_gate_types)
 
 
+def substitute(
+    initial_resources: Resources, name: str, replacement_resources: Resources, in_place=False
+) -> Resources:
+    """Replaces a specified gate in a :class:`~.resource.Resources` object with the contents of another :class:`~.resource.Resources` object.
+
+    Args:
+        initial_resources (Resources): the :class:`~.Resources` object to be modified
+        name (str): the name of the gate to be replaced
+        replacement (Resources): the :class:`~.Resources` containing the resources that will replace the gate
+
```

Comment:
```suggestion
    Args:
        initial_resources (Resources): the resources to be modified
        name (str): the name of the operation to be replaced
        replacement (Resources): the resources to be substituted instead of the gate
        in_place (bool): determines if the initial resources are modified in place or if a new copy is created

```